observer/monitor.go

```diff
@@ -0,0 +1,54 @@
+package observer
+
+import (
+	"time"
+
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/observer/plugins"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type monitor struct {
+	name     string
+	period   time.Duration
+	timeout  time.Duration
+	pluginIs string
+	probe    plugins.Plugin
+	logger   blog.Logger
+	metric   prometheus.Registerer
+}
+
+func (m monitor) start() *time.Ticker {
+	ticker := time.NewTicker(m.period)
+	go func() {
+		for {
+			select {
+			case tick := <-ticker.C:
+				success, took := m.probe.Do(tick, m.timeout)
+				statTotalObservations.WithLabelValues(m.pluginIs, m.name).Add(1)
+				if !success {
+					statTotalErrors.WithLabelValues(m.pluginIs, m.name).Add(1)
+					m.logger.Infof("%s monitor %q failed while taking:=%s", m.pluginIs, m.name, took.String())
+					return
+				}
+				m.logger.Infof("%s monitor %q succeeded while taking:=%s", m.pluginIs, m.name, took.String())
+			}
+		}
+	}()
+	return ticker
+}
+
+func (m monitor) New(c MonConf, log blog.Logger, prom prometheus.Registerer, t int) *monitor {
```

Comment:
Should verify that timeout is shorter than period - or you could automatically infer a timeout from the period!